[Ong Yi sheng Geoff] iP

geoffong11 opened the pull request:
# RatSpeak
>"Use your mind for more important things" - Lincoln

RatSpeak helps to remember all the tasks that you want to complete
It's
- Easy to use
- Fast
- text-based
- **FREE**

All you need to do is
1. Download it from [here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html)
2. double click it
3. Add your tasks
4. Let it manage your tasks for you! :+1:

Features:
- [x] Managing tasks
- [ ] Managing deadlines
- [ ] Managing events

Some basic commands are:
```
todo
deadline
event
mark
unmark
delete
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
Level 0

Commit:
level-0

Commit:
level-1

Commit:
level-2

Commit:
level-3

Commit:
level-4

Commit:
level-5

Commit:
level-6

Commit:
A-TextUiTesting

Commit:
finish the general loading and writing of files

Commit:
finish marking and unmarking of files in storage

Commit:
Merge branch 'branch-level-7'

Commit:
level-8

Commit:
Merge branch 'branch-level-8'

Commit:
add ui

Commit:
more OOP

Commit:
Merge remote-tracking branch 'origin/add-gradle-support'

Commit:
add gradle build

Commit:
packaging

Commit:
give Ui more things to do

Commit:
Merge branch 'javadoc'

Commit:
updating OOP

Commit:
test merge

Commit:
resolved

Commit:
level 9

Commit:
Merge branch 'level-9'

Commit:
updating comments

Commit:
coding standards

Commit:
level 10

Commit:
Merge branch 'branch-level-10'

Commit:
update javadoc and packaging and gui

Commit:
update javadoc

supermii2 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,57 @@
+package duke;
```
I think you should rename duke to your chatbots name!

geoffong11 replied:
Thanks for the input!

supermii2 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,57 @@
+    private final CommandProcessor commandProcessor;
+    private final Ui ui;
+    private final InputReceiver inputReceiver;
+
```
I don't like how many newlines there are here! Please use a maximum of one. This goes for any subsequent times multiple newlines are used in a row.

geoffong11 replied:
Thank you for catching my mistake!

supermii2 commented on src/main/java/duke/InputReceiver.java:
```diff
@@ -0,0 +1,30 @@
+package duke;
```
Does this need to be its own class? Or can it be contained within parser? It is effectively just a scanner class with only 3 methods used after all

geoffong11 replied:
Thank you for your input!

supermii2 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,33 @@
+package duke;
+
+public class Ui {
+    private static final String ANSWER_BORDER = "=========================================";
```
Good use of final to declare constants. Admittedly I forgot about this in my own project

geoffong11 replied:
Thank you!

supermii2 commented on src/main/java/duke/data/TaskList.java:
```diff
@@ -0,0 +1,148 @@
+        int taskIndex = getTaskIndex(taskNumber);
+        Task markedTask = this.tasklist.get(taskIndex).done();
+        tasklist.set(taskIndex, markedTask);
+        STORAGE.writeAllToFile(tasklist);
```
I like how storage is handled in the TaskList logic, and not the command logic!

geoffong11 replied:
Thank you!

supermii2 commented on src/main/java/duke/data/TaskList.java:
```diff
@@ -0,0 +1,148 @@
+    public String mark(String taskNumber) throws DukeException {
+        int taskIndex = getTaskIndex(taskNumber);
+        Task markedTask = this.tasklist.get(taskIndex).done();
+        tasklist.set(taskIndex, markedTask);
```
Is this line necessary? The previous line already set the Task object to marked! There is no need to replace it with itself

geoffong11 replied:
Thanks for your  input!

supermii2 commented on src/main/java/duke/data/TaskList.java:
```diff
@@ -0,0 +1,148 @@
+    private static final Storage STORAGE = new Storage();
+
+    /**
+     * initialize TaskList object
```
For readability, I think it would be better to be consistent in capitalisation in your java docs. That is, either capitalise the first letter at the start of sentences in all of them, or don't. (See line 10).

geoffong11 replied:
Thank you for catching out my mistake!

supermii2 commented on src/main/java/duke/exception/DukeException.java:
```diff
@@ -0,0 +1,8 @@
+package duke.exception;
+
```
Missing JavaDocs!

geoffong11 replied:
Thanks for the reminder!

supermii2 commented on src/main/java/duke/parser/CommandProcessor.java:
```diff
@@ -0,0 +1,109 @@
+
+public class CommandProcessor {
+    private final TaskList tasks;
+    private static final String[] VALIDCOMMANDS = {"mark", "unmark", "list", "todo"
```
I like this! It makes enabling and disabling commands much simpler

geoffong11 replied:
Thanks you!

supermii2 commented on src/main/java/duke/parser/CommandProcessor.java:
```diff
@@ -0,0 +1,109 @@
+    private static final String[] VALIDCOMMANDS = {"mark", "unmark", "list", "todo"
+            , "event", "deadline", "delete", "find"};
+
+    private static final Storage storage = new Storage();
```
I think it's a bit strange for the storage object to be stored under the parser. Perhaps find a way to keep it somewhere like Duke!

geoffong11 replied:
Thanks for your input!

supermii2 commented on src/main/java/duke/parser/CommandProcessor.java:
```diff
@@ -0,0 +1,109 @@
+            String taskName = splitCommand[1];
+
+
+            switch (commandType) {
```
I'm not a fan of linking each command to a corresponding method in TaskList. Perhaps split each command into a new class and limit methods in TaskList to operations that affect the TaskList directly

geoffong11 replied:
Thanks for your input!

supermii2 commented on src/main/java/duke/task/Todo.java:
```diff
@@ -0,0 +1,63 @@
+    private static String parseTodo(String task) throws DukeException {
+        String todoTask = task.trim();
+        if (todoTask.isEmpty()) {
+            throw new DukeException("Please enter valid deadline: Do not leave it empty");
```
I don't understand when this is thrown... A todo task should have no deadline anyway

geoffong11 replied:
Thanks for catching my mistake! It should be todo, and it is thrown when the person types todo but did not give a description.

supermii2 commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,92 @@
+        this.deadlineDate = deadlineDate;
+    }
+
+    /**
```
Is it necessary to return a new object everytime a change is made to the original object?

geoffong11 replied:
Thanks for your input!

supermii2 commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,92 @@
+    private final LocalDate deadlineDate;
+
+
+    private static String[] parseDeadline(String task) throws DukeException {
```
I think this would be better placed in the Parser object! Or at least handled there in some way.

geoffong11 replied:
Thanks for your input! I feel by putting it into the respective class, it will be more modularized and clearer what are the requirements for a deadline.

supermii2 commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,107 @@
+package duke.task;
```
My comments on Deadline.java apply here too!

geoffong11 replied:
Thanks for your input!

supermii2 reviewed (commented):
The readibility is solid! However, it was hard to keep track of the logical flow of the program. I think separating the commands into different classes to handle behavior for each command, and handling storage as part of duke instead of parser would be the logical next step!

Commit:
update javadocs and code based on comments

maj0-0 commented on src/main/java/ratspeak/Duke.java:
```diff
@@ -0,0 +1,43 @@
+import ratspeak.parser.CommandProcessor;
+
+public class Duke {
+
```
You can consider adding Javadoc comments to describe the purpose of this class, otherwise I think Duke class looks good!

maj0-0 commented on src/main/java/ratspeak/InputReceiver.java:
Seems like it follows all coding standards!

maj0-0 commented on src/main/java/ratspeak/data/TaskList.java:
```diff
@@ -0,0 +1,148 @@
+ * performs add, delete, mark, unmark
+ */
+public class TaskList {
+    private ArrayList<Task> tasklist;
```
Consider changing the variable name tasklist to taskList to follow Java naming conventions.

maj0-0 commented on src/main/java/ratspeak/gui/Main.java:
Code appears to follow Java naming conventions and maintains good practices!

maj0-0 commented on src/main/java/ratspeak/storage/Storage.java:
```diff
@@ -0,0 +1,120 @@
+            }
+            return new ArrayList<Task>();
+        } catch(Exception e) {
+            System.out.println(e.getMessage());
```
Instead of printing the exception message directly to the console, maybe consider logging the error, or using a more user-friendly error message to avoid confusion by the user from an error message that's too complicated.

maj0-0 reviewed (commented):
Looks good overall! Code follows standard coding conventions and is consistent throughout! :)

Commit:
update code

Commit:
Update task

Commit:
Update assertions

geoffong11 committed:
Merge pull request #1 from geoffong11/A-Assertions

Update assertions

Commit:
Improve code quality

There are some methods that have multiple levels of abstraction.

Change most of the multiple level abstractions to single level of abstractions

As a step towards further improving code quality, let's abstract more of the
code to achieve more single level abstractions.

geoffong11 committed:
Merge pull request #2 from geoffong11/A-CodeQuality

Improve code quality

Commit:
Add reminder feature to the bot

Within 7 days of a deadline, the bot will add it to its reminders,
showing the information of the tasks when it starts running and
when the user wants to see the list of tasks

Improve code quality for Storage class and the various Task classes

Split long methods into smaller methods to maintain readability. Doing
so will make future editing easier.

Commit:
update javafx to make it better

geoffong11 committed:
Update README.md

Commit:
add ui picture

Commit:
Merge branch 'master' of https://github.com/geoffong11/ip

Commit:
update javadoc

Commit:
update java
